File: app/src/personnages/Ennemi.java
package personnages;

import objets.Exp;
import objets.Objet;

import java.util.ArrayList;

public class Ennemi extends Entite {

    private double poidsSpawn;
    private final Exp xpDrop;

    private final ArrayList<Objet> drops;

    public static final Ennemi LAMBDA = new Ennemi("lambda", 0, 0, 0, 0, 0, new ArrayList<>());


    public Ennemi(String nom, int ptsVie, int niveau, int att, double poidsSpawn, int xpDrop, ArrayList<Objet> drops) {
        super(nom, ptsVie, niveau, att);
        this.drops = drops;
        this.poidsSpawn = poidsSpawn;
        this.xpDrop = new Exp(xpDrop);
    }

    /**
     * Constructeur copie
     * @param autre Objet initial qu'on veut copier
     */
    public Ennemi (Ennemi autre)
    {
        super(autre);
        this.drops = autre.drops;
        this.poidsSpawn = autre.poidsSpawn;
        this.xpDrop = new Exp(autre.xpDrop.getValeur());
    }

    // public void setPoidsSpawn(int poidsSpawn) { this.poidsSpawn = poidsSpawn; }
    public double getPoidsSpawn() { return this.poidsSpawn; }
    public String getNom() { return this.nom; }
    public String toString(){
        return String.format("%s\n\nNiv:%d\tPV:%d\tAtt:%d\t", nom, niveau, ptsVie, attBase);
    }
    public Exp getXpDrop() { return this.xpDrop; }
    public ArrayList<Objet> getDrops ()
    {
        return this.drops;
    }

    public static void pondererPoidsSpawn(ArrayList<Ennemi> ennemis) {
        double totalDropRates = 0;
        for (Ennemi ennemi : ennemis) {
            totalDropRates += ennemi.poidsSpawn;
        }
        for (Ennemi ennemi : ennemis) {
            ennemi.poidsSpawn /= totalDropRates;
        }
    }

}

File: KF-Remake/Kingdom-Fall/source/personnages/Ennemi.cs
using System.Collections;
namespace personnages;

public class Ennemi : Entite 
{
    public double poidsSpawn { get; set; }
    public List<objets.Objet> drops { get; }
    internal objets.Exp xpDrop { get; }

    public static Ennemi LAMBDA { get { return new Ennemi(); } }

    public Ennemi() : base("lambda", 0, 0, 0)
    {
        this.poidsSpawn = 0;
        this.drops = new List<objets.Objet>();
        this.xpDrop = new objets.Exp(0);
    }

    public Ennemi(string nom, int ptsVie, int niveau, int attBase, double poidsSpawn, int xpDrop, List<objets.Objet> drops) : base(nom, ptsVie, niveau, attBase)
    {
        this.drops = drops;
        this.poidsSpawn = poidsSpawn;
        this.xpDrop = new objets.Exp(xpDrop);
    }

    public Ennemi (Ennemi autre) : base(autre)
    {
        this.drops = autre.drops;
        this.poidsSpawn = autre.poidsSpawn;
        this.xpDrop = new objets.Exp(autre.xpDrop.valeur);
    }

    public override string ToString()
    {
        return $"{base.nom}\n\nNiv:{niveau}\tAtt:{attBase}\t";
    }

    public static void PondererPoidsSpawn(List<Ennemi> ennemis)
    {
        double totalDropRates = 0;
        foreach (Ennemi ennemi in ennemis)
        {
            totalDropRates += ennemi.poidsSpawn;
        }
        foreach (Ennemi ennemi in ennemis)
        {
            ennemi.poidsSpawn /= totalDropRates;
        }
    }
}
